src/physics/em/detail/UrbanMscStepLimit.hh

```diff
@@ -105,20 +126,29 @@ UrbanMscStepLimit::UrbanMscStepLimit(const UrbanMscRef&       shared,
                                      GeoTrackView*            geometry,
                                      const PhysicsTrackView&  physics,
                                      const MaterialView&      material,
-                                     const SimTrackView&      sim)
+                                     const SimTrackView&      sim,
+                                     real_type                phys_step)
     : shared_(shared)
     , inc_energy_(particle.energy())
+    , is_positron_(particle.particle_id() == shared.ids.positron)
     , safety_(geometry->find_safety(geometry->pos()))
     , params_(shared.params)
     , msc_(shared_.msc_data[material.material_id()])
-    , helper_(shared, particle, physics, material)
+    , helper_(shared, particle, physics)
     , on_boundary_(sim.num_steps() == 0 || safety_ == 0)
+    , phys_step_(phys_step)
 {
     CELER_EXPECT(particle.particle_id() == shared.ids.electron
                  || particle.particle_id() == shared.ids.positron);
+    CELER_EXPECT(phys_step > 0);
 
+    // Mean slowing-down distance from current energy to zero
     range_  = helper_.range();
+    // Mean free path for MSC at current energy
     lambda_ = helper_.msc_mfp(inc_energy_);
+
+    // The slowing down range should already have been applied as a step limit
+    CELER_ENSURE(range_ >= phys_step_);
```

Comment:
This assumes that the eloss process (range limit) and the msc process are always come together.  In general, this may be true with a normal physics list (set of processes), but may not support other special test cases (i.e, msc studies without the eloss range limit or msc with a biasing even though I cannot think of a good case).

Reply:
Fair enough; but I'd rather design for the standard use case than to have to keep track of a bunch of exceptional/unusual cases. For test purposes, wouldn't "MSC studies without the eloss range limit" just give the same thing as studies with the range limit, due to the extra range check that was previously built in here? I think it's way better to add explicit assertions about the expected behavior than riddle the code with `a = min(a, limit)` that are effectively null-ops in the real transport code. It's much easier to reason about (and test!) the expected behavior of the code without having to worry about such exceptions.